sim/firesim-lib/src/main/scala/bridges/TestFinisherBridge.scala

```diff
@@ -0,0 +1,97 @@
+
+package firesim.bridges
+
+import chisel3._
+import chisel3.util._
+import freechips.rocketchip.config.Parameters
+import midas.widgets._
+import midas.widgets.PeekPokeBridge
+
+case class CodeElaborator(isErr: Boolean, message : String)
+
+case class TestIndicatorParams(
+  nConcurrentTests: Int     = 1,
+  codeMessage: Seq[CodeElaborator]
+) {
+  def codeBits: Int    = 16
```

Comment:
If you want to keep this around: 
```suggestion
  def codeBits: Int    = log2ceil(codeMessage.length)
```